Guide/jobs.markdown

```diff
@@ -91,3 +91,178 @@ We can then create a cron entry such as
 to schedule a job to be run every day at 5am. See [this article by DigitalOcean](https://www.digitalocean.com/community/tutorials/how-to-use-cron-to-automate-tasks-ubuntu-1804) for more information on `cron`.
 
 
+## Jobs Dashboard
+
+IHP comes with a built in jobs dashboard for running and viewing your jobs. To define which jobs
+you want displayed in the dashboard, you construct a custom JobDashboard type that IHP will use to determine the dashboard's behavior.
+
+### Quick Start
+
+```haskell
+import IHP.Job.Dashboard
+
+type MyJobDashboard = JobsDashboardController
+    (BasicAuthStatic "jobs" "jobs")
+    [UpdateUserJob, CleanDatabaseJob, OtherRandomJob]
```

Comment:
Could we make this zero-config out of the box? E.g. we could detect all job queue tables automatically by looking for tables with the `_jobs` suffix :)

Reply:
I don't believe that would be possible with the current design. Since it's all type level I'm not sure how we could dynamically search for table names and get their corresponding types...I'm sure it's possible but nothing is coming to mind atm.

Also, since it gets injected to the user's application and not the IDE, I think adding it to the project by default might have some negative security implications

Reply:
> Since it's all type level I'm not sure how we could dynamically search for table names and get their corresponding types...I'm sure it's possible but nothing is coming to mind atm.

Yes, I think when we want to do this will all tables, we will loose a bit of type safety as we have to check everything at runtime. Basically it would always pass around table names as a string. But this might make the code even a bit simpler.

E.g. the IHP data editor is also using a similiar approach. It's also working with a dynamic table schema which it detects at runtime.

I think this is worth to explore as otherwise you'll always need to manually add your jobs to the dashboard type whenever you add a new job.

> Also, since it gets injected to the user's application and not the IDE, I think adding it to the project by default might have some negative security implications

I'd still keep the required configuration for the authentication. This way it's still safe :)

Reply:
> we will loose a bit of type safety as we have to check everything at runtime

I'm not so much worried about the type safety, but rather the loss of functionality that doing this dynamically would come with. By having to specify types, users can (and pretty much _will_, in order for this to be that useful) define custom implementations of the `DisplayableJob`. This allows us to include fields like parent objects (i.e. in Attics, the ScrapeBandJob has a band_id, and its custom implementation fetches the band and displays it in the table).

I found this function [`someSymbolVal`](https://hackage.haskell.org/package/base-4.14.1.0/docs/GHC-TypeLits.html#v:someSymbolVal) which if I'm understanding it correctly allows you to promote a String to a Symbol at runtime. I will look into this, so hopefully we can get the best of both worlds :)